applications/StructuralMechanicsApplication/custom_constitutive/linear_j2_plasticity_3d.cpp

```diff
@@ -0,0 +1,464 @@
+                    delta_k * (mAccumulatedPlasticStrain -
+                            (1/hardening_exponent) * (1- std::exp(-hardening_exponent * mAccumulatedPlasticStrain)));
+    return Wp_new;
+}
+
+double LinearJ2Plasticity3D::GetDeltaGamma(double norm_s_trial,
+                                           const Properties& rMaterialProperties)
+{
+    double E = rMaterialProperties[YOUNG_MODULUS];
+    double poisson_ratio = rMaterialProperties[POISSON_RATIO];
+    double yield_stress = rMaterialProperties[YIELD_STRESS];
+    double theta = rMaterialProperties[REFERENCE_HARDENING_MODULUS];
+    double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+    double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+    double tolerance = 1e-6 * yield_stress;
+    double mu = E / (2. * (1. + poisson_ratio));
+    double dgamma = 0.0;
+    double norm_yieldfunction = 1.0;
+
+    while (norm_yieldfunction > tolerance)
+    {
+        double k_new = GetSaturationHardening(rMaterialProperties);
+        double kp_new =
+            theta * hardening_modulus +
+            delta_k * (hardening_exponent *
+                       std::exp(-hardening_exponent * mAccumulatedPlasticStrain));
+        double yieldfunction = -0.8164965809277260 * k_new + norm_s_trial - 2. * mu * dgamma;
+        double derivative_yieldfunction = -2. * mu * (1. + kp_new / (3. * mu));
+        dgamma = dgamma - yieldfunction / derivative_yieldfunction;
+        mAccumulatedPlasticStrain =
+            mAccumulatedPlasticStrainOld + 0.8164965809277260 * dgamma;
+        norm_yieldfunction = std::abs(yieldfunction);
+    }
+    // TODO (marcelo): handle the case when no convergence is achieved.
+    return dgamma;
+}
+
+double LinearJ2Plasticity3D::yieldFunction(const double norm_dev_stress,
+                                           const Properties& rMaterialProperties)
+{
+    double yield_stress = rMaterialProperties[YIELD_STRESS];
+    double hardening_modulus = rMaterialProperties[ISOTROPIC_HARDENING_MODULUS];
+    double theta = rMaterialProperties[REFERENCE_HARDENING_MODULUS];
+    double delta_k = rMaterialProperties[INFINITY_HARDENING_MODULUS];
+    double hardening_exponent = rMaterialProperties[HARDENING_EXPONENT];
+    double k_old =
+        yield_stress + (theta * hardening_modulus * mAccumulatedPlasticStrainOld) +
+        (delta_k) * (1. - std::exp(-hardening_exponent * mAccumulatedPlasticStrainOld));
+
+    return norm_dev_stress - k_old * 0.8164965809277260; // sqrt(2/3)
+}
+
+void LinearJ2Plasticity3D::CalculateElasticMatrix(Matrix &D, const Properties &props)
+{
+    double E = props[YOUNG_MODULUS];
+    double poisson_ratio = props[POISSON_RATIO];
+    double lambda =
+        E * poisson_ratio / ((1. + poisson_ratio) * (1. - 2. * poisson_ratio));
+    double mu = E / (2. + 2. * poisson_ratio);
+
+    D.clear();
```

Comment:
I think that if you do a Clear and a resize with false your are emptying the matrixes twice

Reply:
@RiccardoRossi Can you confirm?